code/modules/mob/living/simple_animal/friendly/pet.dm

```diff
@@ -42,7 +42,12 @@
 		regenerate_icons()
 
 /mob/living/simple_animal/pet/Destroy()
-	QDEL_NULL(pcollar)
+	if(pcollar)
+		pcollar.forceMove(drop_location())
+		pcollar = null
+	if(access_card)
+		access_card.forceMove(drop_location())
+		access_card = null
```

Comment:
Could you call this stuff on `death()` instead? A direct qdel generally means chasm or supermatter, which should not leave any remnants.

Reply:
Yes. Destroy() should never leave anything behind.